api/core/runners/workflowai/workflowai_runner.py

```diff
@@ -418,27 +422,13 @@ async def _extract_image_options(
     ) -> ImageOptions | None:
         """Extract the image options from the input and remove it from the input if possible"""
         base = self.properties.image_options
-        if self._prepared_output_schema.min_file_count:
-            if not base:
-                base = ImageOptions()
-            if not base.image_count:
-                base.image_count = self._prepared_output_schema.min_file_count
-
-        # We consider that input can contain the same fields as ImageOptions
-        try:
-            extracted_options = ImageOptions.model_validate(input)
-        except ValidationError:
-            return base
-
-        # In case of validation, we can check which key was actually extracted
-        extracted_keys = extracted_options.model_dump(exclude_unset=True)
 
-        if delete_keys:
-            # We can remove the corresponding fields from the input since they are already used
-            for key in extracted_keys:
-                input.pop(key, None)
+        if base is None:
```

Comment:
@guillaq without that the image count is None, and this code is skipped:

`if messages[0].image_options and messages[0].image_options.image_count:
            generation_config.responseModalities = ["IMAGE", "TEXT"]`

Reply:
I'll let you see what you do with that, but I think there is no "image_options" in the input schema right ?